tests/integration/actions/test_snap_installer.py

```diff
@@ -0,0 +1,142 @@
+# Copyright 2021 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Tests for snap installer."""
+import os
+import pathlib
+import shutil
+import subprocess
+import sys
+
+import pytest
+
+from craft_providers.actions import snap_installer
+
+
+@pytest.fixture()
+def empty_test_snap(tmp_path):
+    """Fixture to provide an empty local-only snap for test purposes.
+
+    Requires:
+    CRAFT_PROVIDERS_TESTS_ENABLE_SNAP_INSTALL=1
+    """
+    snap_name = "craft-integration-test-snap"
+
+    if shutil.which("snap") is None or sys.platform != "linux":
+        pytest.skip("requires linux and snapd")
+
+    if not os.environ.get("CRAFT_PROVIDERS_TESTS_ENABLE_SNAP_INSTALL") == "1":
+        pytest.skip("need permission to install snaps, skipped")
+
+    if pathlib.Path("/snap", snap_name).exists():
+        pytest.skip(f"test snap {snap_name} already exists, please remove")
```

Comment:
You could potentially make this a fixture and reduce the copy/paste of this structure. Different PR should be fine

Reply:
Will do.

Reply:
but where?

Reply:
I'm thinking I can do this by updating the installed_snap() fixture to take an optional `try_dir` kwarg to install a snap from a directory using try.

Reply:
Done.